packages/drivers/odsp-driver/src/odspDocumentStorageManager.ts

```diff
@@ -459,12 +352,261 @@ export class OdspDocumentStorageService implements IDocumentStorageService {
         });
     }
 
-    private async fetchSnapshot(snapshotOptions: string, tokenFetchOptions: TokenFetchOptions) {
+    private isTreesLatestSecondCallEvent(tokenFetchOptions: TokenFetchOptions) {
+        if (tokenFetchOptions.refresh) {
+            // This is the most critical code path for boot.
+            // If we get incorrect / expired token first time, that adds up to latency of boot
+            this.logger.sendErrorEvent({ eventName: "TreeLatest_SecondCall", hasClaims: !!tokenFetchOptions.claims });
+        }
+    }
+
+    private async fetchSnapshotWithMultiPartRequest(
```

Comment:
Nit: From a naming perspective, can we call this fetchSnapshotViaPost, instead of MultiPart, etc. which is too much implementation detail (that can change)?
Just a thought, definitely not blocking this PR either way.

Reply:
I think fetchSnapshotViaPost is good.